services/src/modules/authentication/get-secret.ts

```diff
@@ -10,22 +11,43 @@ const createClient = (jwks as unknown) as (options: ClientOptions) => JwksClient
 
 const jwkClients: Record<string, JwksClient> = {};
 
-function getJwkClient(jwksUri: string) {
-  if (!jwkClients[jwksUri]) {
+async function getJwksUri(authority: string) {
+  try {
+    const discoveryResponse = await fetch(`${authority}/.well-known/openid-configuration`);
+    const discoveryResult = await discoveryResponse.json();
+
+    return discoveryResult['jwks_uri'] as string;
+  } catch {
+    return;
+  }
+}
+
+const defaultConfig: Partial<ClientOptions> = {
+  cacheMaxAge: 24 * 60 * 60 * 1000,
+};
+
+async function getJwkClient(authority: string, jwksConfig: Partial<ClientOptions>) {
+  if (!jwkClients[authority]) {
+    const jwksUri = await getJwksUri(authority);
```

Comment:
If `jwksConfig` contains `jwksUri`, it will override the one we retrieve here from the authority (as it should).
Probably worth checking if the config contains it before retrieving it here, it will save a redundant call to the authority.

Reply:
You're right. I've fixed